src/recipes/ios-promo/index.js

```diff
@@ -0,0 +1,284 @@
+  strict:true, undef:true, curly:false, browser:true,
+  unused:true, indent:2, maxerr:50, devel:true, node:true, boss:true, white:true,
+  globalstrict:true, nomen:false, newcap:true, esnext: true, moz: true  */
+
+/*global require, exports */
+
+
+"use strict";
+let common     = require("../../common");
+let configFile = require("./config");
+let actions    = common.actions;
+let log        = actions.log.bind(actions.log, "pb-mode-survey");
+
+let uuid       = require("node-uuid").v4;
+let events     = require("../../common/events");
+
+let { hasAny } = require("../../jetpack/array");
+
+let UITour     = require("thirdparty/uitour");
+
+let { Flow, phonehome } = require("../../common/heartbeat/");
+let { Lstore } = require("../../common/recipe-utils");
+
+// TODO, is it module level config, or function level?
+// so far, 3 systems
+// - module.config
+// - function config
+// - external 'sampling config'
+
+/*
+  Full description:
+
+  - Should wait some period between requests
+  - will only get people on the first impression (initial startup of a session)
+
+     - no try
+     - no 24-ish
+
+  - different sample percentages for differnt channels.
+  - different 'wait between' guarantees?
+
+*/
+let days = 24 * 60 * 60 * 1000;
+
+
+//Gets a random branch (even distribution)
+let chooseBranch = function(branches, rng=Math.random() ) {
+  let num_branches = branches.length;
+  for (var i = 1; i <= num_branches ; i++) {
+    if (rng < (i/(num_branches))) {
+      return branches[i-1];
+    }
+  }
+  return branches[num_branches-1];
+};
+
+const BRANCH        = chooseBranch(configFile.studies);
+const NAME          = BRANCH.key;
+const VARIATION     = BRANCH.name;
+const VERSION       = BRANCH.version;
+const DELAY         = BRANCH.delay;
+const KEY           = BRANCH.key;  //  'ios-promo'.  keys localstore
+//const PHONEHOMEPCT = configFile.study.phonehomepct
+
+let config = {
+  lskey:     KEY,
+  survey_id: KEY
+};
+
+// setup state?
+
+/*eData = {
+  data: {
+    flows: {},
+    lastRun: number
+  }
+},
+*/
+
+var setupState = function (key, storage) {
+  key     = key || config.lskey;
+  storage = storage || localStorage;
+
+  var eData = new Lstore(key, storage).revive().store();  // create or revive
+
+  if (! eData.data.flows)   eData.data.flows   = {};
+  if (! eData.data.lastRun) eData.data.lastRun = 0;
+
+  eData.store();
+
+  return eData;
+};
+
+// module level
+var eData = setupState();
+
+// ELIGIBLE.  parts of the eligibility, made explicit for testability
+
+// is dayspassed >= restDays
+let waitedEnough = function (restDays, last, now) {
+  now = now || Date.now();
+  let dayspassed = ((now - last)/days);
+  return dayspassed >= restDays ;
+};
+
+
+// OH BOY.  This is scary :)
+let isAustralia = function () {
+  var current_date = new Date( );
+  var gmt_offset = current_date.getTimezoneOffset( ) / 60;
+  return  ( -11 < gmt_offset ) && (gmt_offset <= -7  )
```

Comment:
Isn't it actually "< -7" for aus.  Minor point since it's the only country there that we care about.